### PR TITLE
[T3A5][W13-A1] Conan Kian

### PR DESCRIPTION
**Pull request by ckjr**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/ui/Formatter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    
+    /** A decorative prefix added to the beginning of lines printed by AddressBook */
+    private static final String LINE_PREFIX = "|| ";
+    
+    /** A platform independent line separator. */
+    private static final String LS = System.lineSeparator();
+
+    private static final String DIVIDER = "===================================================";
+    
+    private final PrintStream out = System.out;
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: errr.. why is this here?

File: src/seedu/addressbook/ui/Formatter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    
+    public void print(String arg) {
+        showToUser(arg);
+    }
+    
+    public void printWithPrefix(String arg) {
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: Formatter is doing 2 things now
1. formatting
2. printing
Essentially, you have made changes to violate SRP in a different way ;-)

File: src/seedu/addressbook/ui/TextUi.java [hunk 3]
```diff
@@ -76,46 +64,34 @@ private boolean isCommentLine(String rawInputLine) {
      * @return command (full line) entered by the user
      */
     public String getUserCommand() {
-        out.print(LINE_PREFIX + "Enter command: ");
+        formatter.printWithPrefix("Enter command: ");
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/TextUi.java]: you should be invoking the `Formatter` here to provide you the formatted string. Displaying the string is still the UI's responsibility.

**Comment by okkhoy** [comment]
@ckjr 
Some comments added. Please ack & close the PR after reading comments.

Additionally, 
Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

**Comment by okkhoy** [comment]
@ckjr 
Some comments added. Please ack & close the PR after reading comments.

Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/ui/Formatter.java (hunk 1 above)
Comment (ckjr) [inline src/seedu/addressbook/ui/Formatter.java]: Kind of misunderstood SRP. What I am supposed to do in Formatter is simply process and _return_ the text right? I should not have transferred `showToUser` to the Formatter class, am I right?

**Comment by ckjr** [comment]
Acknowledged
